Add current key of the ill-thought-out localhost.direct project

wklaebe opened the pull request.

BenBE commented:
Some minor additions:

1. document the password as `password = DoNotPutKeyInPublicAccessiblePath` in the meta file.
2. Note the naming convention regarding certificates.
3. Only the leaf certificate should be put in the `.crt` file
4. Rename the files to indicate some validity period to allow for further certificates to be added
5. Speaking of adding further certificates:

```
-----BEGIN RSA PRIVATE KEY-----
MIIEowIBAAKCAQEAv6nXuothaMz3kOl1pAAkvUio98MdAQBbHK+q81M5Q+9MTM+e
Eah5W4hR2U56a7NHIJntYA+TbENpz0VWkm91Cn8VBxtzXXblNj9R9OPhEFkaZ1Fs
lWFwt1BG6Y6bc7EmgDEEidpFQpbyccm+oWRh4oH2m+lUYO1LJtzz8m7VHtae9NBy
Dpq1U/nxaM69KWeIUeMskJ7Yggmwn5talkeuqMwSURa4BmrWXSk/XYWhA8BSl2c2
w1uDzzHAtvWSSWC+C25TvW5o77val4l28oOo/OmR6LnOr/5vpfo4cH1qA0GO5va9
SqkkLWj1bKSjVYlQf3TzIUYuhwZuJ8KCuho56QIDAQABAoIBAECO+g5owm/lkoIq
USjjr3xejCOgghWDRC1ecouxfbmHUkERMeIqyG3vdfpnd87tPUka8B0KQhMWhf5+
EZITnu3hLnkNYweZc82Y+IcvdvDlwoIIqDWM88WbcbdcBLqschZ97FRuxCOQ6qsv
VFnSkTIVqraLmpNneoflpvoDISBYfz9CS7jpqWpkschjOk5DsOuPMVoDTgAk6kaE
QpNyJFutNUdlDdNVXQy0xcNhJTuPj5GwahcLAz5HkZ6QNNXVCvJ8kyrCeR0cVHrR
ZkevwZaFwNP2gUVulhoJkOfabz2XJPvTntZWU0YPDF7wDUsIAr3HIdpigBO4r68z
BOtwzAECgYEA4wpZo7YjJPc7dKZBOVq6RgIZ9GASgcZD16uOKH61ZJvPVHHZbuB8
mh+FOB2YMBxWT59It3EiFM1xHFpwYpCPYnxK+upuAoNueCLAayYHQU4JcXJ5SE02
o78iRwpkahVuW5q2ZFfxgu2XbSHHRO/cCo8PU03WMSOyRsx2Y8NGVoECgYEA2BxQ
A832L5YK8XU5mpDyaKXbOvtqXxzfwvL6ejFmni0dkqTGotJrbiQhOOTb0X9SyBnO
4Ebnl2scpA7wgqxyBc5YqV82Hjnczq0T5wkr91pdjdW8iel6sdAv6sucwwWs2seT
ndhp74X/rstIeXCk2E4Zzvv22wYmpKB8yp13P2kCgYEAjKak0Z7MNuJSmvlD/twN
+cDjg1jhHjRNYxaMgtm8Vg+Qaaf+I4o+ptxQE2uCFIv5jGoJCluz2V9Ma1w7s1QK
/+AwMzF35Wa20QX2Xmq4o+RMRVO9fuUWrGqEOa5edrnAHE/EHnW80HeQD7OVV0eg
8Yme3OSjhN2wW9vem/8lXAECgYARbkct52Dh+bIsQ44/300OD8KdAUZVIZkSulJl
pLJoM69c8plq7JnpqlRbqGnqfeCJbzt2b4GifusyeFoB0DLqMPiJwLo2uegLa35d
dinrq4g8Nm8cmI5i5S3YPFAKktycUNihNKFmxmIXESJpWSKSp8oBHeWNSPO495DZ
LtY9cQKBgC9VqxlaTshEuCLLC1/iaZouAXvUVNqhxwXYrRWs2/wq6P11J+nkHHHZ
yR9Qie6jA2atsJwjW8OWd1GTnifawzCkW5QZGpVYWTJ0mDjPuZuIfEngIaN5JiL7
XccvtmWqMk50y0UwK8qrYmP0fycrOwZko+TzxhYopcU7SLR88pk5
-----END RSA PRIVATE KEY-----
```

```
-----BEGIN PRIVATE KEY-----
MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC/qde6i2FozPeQ
6XWkACS9SKj3wx0BAFscr6rzUzlD70xMz54RqHlbiFHZTnprs0cgme1gD5NsQ2nP
RVaSb3UKfxUHG3NdduU2P1H04+EQWRpnUWyVYXC3UEbpjptzsSaAMQSJ2kVClvJx
yb6hZGHigfab6VRg7Usm3PPybtUe1p700HIOmrVT+fFozr0pZ4hR4yyQntiCCbCf
m1qWR66ozBJRFrgGatZdKT9dhaEDwFKXZzbDW4PPMcC29ZJJYL4LblO9bmjvu9qX
iXbyg6j86ZHouc6v/m+l+jhwfWoDQY7m9r1KqSQtaPVspKNViVB/dPMhRi6HBm4n
woK6GjnpAgMBAAECggEAQI76DmjCb+WSgipRKOOvfF6MI6CCFYNELV5yi7F9uYdS
QREx4irIbe91+md3zu09SRrwHQpCExaF/n4RkhOe7eEueQ1jB5lzzZj4hy928OXC
ggioNYzzxZtxt1wEuqxyFn3sVG7EI5Dqqy9UWdKRMhWqtouak2d6h+Wm+gMhIFh/
P0JLuOmpamSxyGM6TkOw648xWgNOACTqRoRCk3IkW601R2UN01VdDLTFw2ElO4+P
kbBqFwsDPkeRnpA01dUK8nyTKsJ5HRxUetFmR6/BloXA0/aBRW6WGgmQ59pvPZck
+9Oe1lZTRg8MXvANSwgCvcch2mKAE7ivrzME63DMAQKBgQDjClmjtiMk9zt0pkE5
WrpGAhn0YBKBxkPXq44ofrVkm89Ucdlu4HyaH4U4HZgwHFZPn0i3cSIUzXEcWnBi
kI9ifEr66m4Cg254IsBrJgdBTglxcnlITTajvyJHCmRqFW5bmrZkV/GC7ZdtIcdE
79wKjw9TTdYxI7JGzHZjw0ZWgQKBgQDYHFADzfYvlgrxdTmakPJopds6+2pfHN/C
8vp6MWaeLR2SpMai0mtuJCE45NvRf1LIGc7gRueXaxykDvCCrHIFzlipXzYeOdzO
rRPnCSv3Wl2N1byJ6Xqx0C/qy5zDBazax5Od2Gnvhf+uy0h5cKTYThnO+/bbBiak
oHzKnXc/aQKBgQCMpqTRnsw24lKa+UP+3A35wOODWOEeNE1jFoyC2bxWD5Bpp/4j
ij6m3FATa4IUi/mMagkKW7PZX0xrXDuzVAr/4DAzMXflZrbRBfZearij5ExFU71+
5RasaoQ5rl52ucAcT8QedbzQd5APs5VXR6DxiZ7c5KOE3bBb296b/yVcAQKBgBFu
Ry3nYOH5sixDjj/fTQ4Pwp0BRlUhmRK6UmWksmgzr1zymWrsmemqVFuoaep94Ilv
O3ZvgaJ+6zJ4WgHQMuow+InAuja56Atrfl12KeuriDw2bxyYjmLlLdg8UAqS3JxQ
2KE0oWbGYhcRImlZIpKnygEd5Y1I87j3kNku1j1xAoGAL1WrGVpOyES4IssLX+Jp
mi4Be9RU2qHHBditFazb/Cro/XUn6eQccdnJH1CJ7qMDZq2wnCNbw5Z3UZOeJ9rD
MKRblBkalVhZMnSYOM+5m4h8SeAho3kmIvtdxy+2ZaoyTnTLRTAryqtiY/R/Jys7
BmSj5PPGFiilxTtItHzymTk=
-----END PRIVATE KEY-----
```

Both taken from commit 58fb13ba348ec47c7c8dcc1522a96632a452ac22 on 2019-02-24 (Global Sign). The certs are given as a chain in that commit too.

hannob commented:
I'd prefer not having any special chars (;, *) in the filenames, makes it easier to deal with for badkeys.

wklaebe commented:
>     1. document the password as `password = DoNotPutKeyInPublicAccessiblePath` in the meta file.

The password is not on the key itself, but only on the downloaded Zip file. After unpacking it is not needed anymore, I'd consider including it to be misleading.

>     2. Note the naming convention regarding certificates.

renamed to localhost.direct.crt

>     3. Only the leaf certificate should be put in the `.crt` file

removed chain

>     4. Rename the files to indicate some validity period to allow for further certificates to be added

renamed to localhost.direct.20241101.*, using the Validity Not Before date.

>     5. Speaking of adding further certificates:
> 
> 
> ```
> -----BEGIN RSA PRIVATE KEY-----
> MIIEowIBAAKCAQEAv6nXuothaMz3kOl1pAAkvUio98MdAQBbHK+q81M5Q+9MTM+e
> Eah5W4hR2U56a7NHIJntYA+TbENpz0VWkm91Cn8VBxtzXXblNj9R9OPhEFkaZ1Fs
> lWFwt1BG6Y6bc7EmgDEEidpFQpbyccm+oWRh4oH2m+lUYO1LJtzz8m7VHtae9NBy
> Dpq1U/nxaM69KWeIUeMskJ7Yggmwn5talkeuqMwSURa4BmrWXSk/XYWhA8BSl2c2
> w1uDzzHAtvWSSWC+C25TvW5o77val4l28oOo/OmR6LnOr/5vpfo4cH1qA0GO5va9
> SqkkLWj1bKSjVYlQf3TzIUYuhwZuJ8KCuho56QIDAQABAoIBAECO+g5owm/lkoIq
> USjjr3xejCOgghWDRC1ecouxfbmHUkERMeIqyG3vdfpnd87tPUka8B0KQhMWhf5+
> EZITnu3hLnkNYweZc82Y+IcvdvDlwoIIqDWM88WbcbdcBLqschZ97FRuxCOQ6qsv
> VFnSkTIVqraLmpNneoflpvoDISBYfz9CS7jpqWpkschjOk5DsOuPMVoDTgAk6kaE
> QpNyJFutNUdlDdNVXQy0xcNhJTuPj5GwahcLAz5HkZ6QNNXVCvJ8kyrCeR0cVHrR
> ZkevwZaFwNP2gUVulhoJkOfabz2XJPvTntZWU0YPDF7wDUsIAr3HIdpigBO4r68z
> BOtwzAECgYEA4wpZo7YjJPc7dKZBOVq6RgIZ9GASgcZD16uOKH61ZJvPVHHZbuB8
> mh+FOB2YMBxWT59It3EiFM1xHFpwYpCPYnxK+upuAoNueCLAayYHQU4JcXJ5SE02
> o78iRwpkahVuW5q2ZFfxgu2XbSHHRO/cCo8PU03WMSOyRsx2Y8NGVoECgYEA2BxQ
> A832L5YK8XU5mpDyaKXbOvtqXxzfwvL6ejFmni0dkqTGotJrbiQhOOTb0X9SyBnO
> 4Ebnl2scpA7wgqxyBc5YqV82Hjnczq0T5wkr91pdjdW8iel6sdAv6sucwwWs2seT
> ndhp74X/rstIeXCk2E4Zzvv22wYmpKB8yp13P2kCgYEAjKak0Z7MNuJSmvlD/twN
> +cDjg1jhHjRNYxaMgtm8Vg+Qaaf+I4o+ptxQE2uCFIv5jGoJCluz2V9Ma1w7s1QK
> /+AwMzF35Wa20QX2Xmq4o+RMRVO9fuUWrGqEOa5edrnAHE/EHnW80HeQD7OVV0eg
> 8Yme3OSjhN2wW9vem/8lXAECgYARbkct52Dh+bIsQ44/300OD8KdAUZVIZkSulJl
> pLJoM69c8plq7JnpqlRbqGnqfeCJbzt2b4GifusyeFoB0DLqMPiJwLo2uegLa35d
> dinrq4g8Nm8cmI5i5S3YPFAKktycUNihNKFmxmIXESJpWSKSp8oBHeWNSPO495DZ
> LtY9cQKBgC9VqxlaTshEuCLLC1/iaZouAXvUVNqhxwXYrRWs2/wq6P11J+nkHHHZ
> yR9Qie6jA2atsJwjW8OWd1GTnifawzCkW5QZGpVYWTJ0mDjPuZuIfEngIaN5JiL7
> XccvtmWqMk50y0UwK8qrYmP0fycrOwZko+TzxhYopcU7SLR88pk5
> -----END RSA PRIVATE KEY-----
> ```
> 
> ```
> -----BEGIN PRIVATE KEY-----
> MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC/qde6i2FozPeQ
> 6XWkACS9SKj3wx0BAFscr6rzUzlD70xMz54RqHlbiFHZTnprs0cgme1gD5NsQ2nP
> RVaSb3UKfxUHG3NdduU2P1H04+EQWRpnUWyVYXC3UEbpjptzsSaAMQSJ2kVClvJx
> yb6hZGHigfab6VRg7Usm3PPybtUe1p700HIOmrVT+fFozr0pZ4hR4yyQntiCCbCf
> m1qWR66ozBJRFrgGatZdKT9dhaEDwFKXZzbDW4PPMcC29ZJJYL4LblO9bmjvu9qX
> iXbyg6j86ZHouc6v/m+l+jhwfWoDQY7m9r1KqSQtaPVspKNViVB/dPMhRi6HBm4n
> woK6GjnpAgMBAAECggEAQI76DmjCb+WSgipRKOOvfF6MI6CCFYNELV5yi7F9uYdS
> QREx4irIbe91+md3zu09SRrwHQpCExaF/n4RkhOe7eEueQ1jB5lzzZj4hy928OXC
> ggioNYzzxZtxt1wEuqxyFn3sVG7EI5Dqqy9UWdKRMhWqtouak2d6h+Wm+gMhIFh/
> P0JLuOmpamSxyGM6TkOw648xWgNOACTqRoRCk3IkW601R2UN01VdDLTFw2ElO4+P
> kbBqFwsDPkeRnpA01dUK8nyTKsJ5HRxUetFmR6/BloXA0/aBRW6WGgmQ59pvPZck
> +9Oe1lZTRg8MXvANSwgCvcch2mKAE7ivrzME63DMAQKBgQDjClmjtiMk9zt0pkE5
> WrpGAhn0YBKBxkPXq44ofrVkm89Ucdlu4HyaH4U4HZgwHFZPn0i3cSIUzXEcWnBi
> kI9ifEr66m4Cg254IsBrJgdBTglxcnlITTajvyJHCmRqFW5bmrZkV/GC7ZdtIcdE
> 79wKjw9TTdYxI7JGzHZjw0ZWgQKBgQDYHFADzfYvlgrxdTmakPJopds6+2pfHN/C
> 8vp6MWaeLR2SpMai0mtuJCE45NvRf1LIGc7gRueXaxykDvCCrHIFzlipXzYeOdzO
> rRPnCSv3Wl2N1byJ6Xqx0C/qy5zDBazax5Od2Gnvhf+uy0h5cKTYThnO+/bbBiak
> oHzKnXc/aQKBgQCMpqTRnsw24lKa+UP+3A35wOODWOEeNE1jFoyC2bxWD5Bpp/4j
> ij6m3FATa4IUi/mMagkKW7PZX0xrXDuzVAr/4DAzMXflZrbRBfZearij5ExFU71+
> 5RasaoQ5rl52ucAcT8QedbzQd5APs5VXR6DxiZ7c5KOE3bBb296b/yVcAQKBgBFu
> Ry3nYOH5sixDjj/fTQ4Pwp0BRlUhmRK6UmWksmgzr1zymWrsmemqVFuoaep94Ilv
> O3ZvgaJ+6zJ4WgHQMuow+InAuja56Atrfl12KeuriDw2bxyYjmLlLdg8UAqS3JxQ
> 2KE0oWbGYhcRImlZIpKnygEd5Y1I87j3kNku1j1xAoGAL1WrGVpOyES4IssLX+Jp
> mi4Be9RU2qHHBditFazb/Cro/XUn6eQccdnJH1CJ7qMDZq2wnCNbw5Z3UZOeJ9rD
> MKRblBkalVhZMnSYOM+5m4h8SeAho3kmIvtdxy+2ZaoyTnTLRTAryqtiY/R/Jys7
> BmSj5PPGFiilxTtItHzymTk=
> -----END PRIVATE KEY-----
> ```
> 
> Both taken from commit 58fb13ba348ec47c7c8dcc1522a96632a452ac22 on 2019-02-24 (Global Sign). The certs are given as a chain in that commit too.

Those two private keys are actually the same, `diff -u <(openssl pkey -text -in localhost.direct.key) <(openssl pkey -text -in localhost.direct.key.pem)` returns no differences.

Added key and certificate as localhost.direct.20190224.*

hannob commented:
I think this can be merged (?)
